### ISSUE TITLE
`CMK_LBDB_ON` usages should be cleaned up

### ISSUE THREAD

**Issue by rbuch** [UIUC-PPL/charm#2549]
Currently, preprocessor directives checking `CMK_LBDB_ON` are overused, often both on the caller and callee side of functions, for example, so they should be consolidated. We should also investigate if this can be reduced to a single (or very small) set of checks that either completely enables or disables the LB framework.

**Comment by stwhite91** [comment]
Should this be closed?

**Comment by rbuch** [comment]
> Should this be closed?

No, it still needs to be cleaned up more, but it won't change functionality at all, so retargeting to 6.12.
